contents/blog/open-source-telemetry-ethical.md

```diff
@@ -0,0 +1,102 @@
+
+From a utilitarian perspective, we believe that tracking the minimum data possible to build useful technology, in a way that doesn't share that data with 3rd parties, means that you have the right approach to product analytics in your open source project.
+
+From a duty-based perspective, we'd argue that transparency is key to covering intent. Since most people won't assume they're being tracked, that means being upfront in the docs and not relying on people digging through your code.
+
+## Sorting the good from the questionable
+
+First of all, open source is exactly that. That's a big step up from SAAS - you can immediately inspect the code or run it on a computer with no internet connection if needed. However, the precise function of the telemetry has user implications.
+
+### What is being tracked
+
+The first step is to think about how much, if anything, a project really needs to track.
+
+#### Nothing
+
+Nada, zilch. 
+
+If an open source project exists for fun, for one person only, or to get better at a particular technology, then we don't think putting extra tracking in makes sense.
+
+In practice, however, it's pretty much impossible not to track more than you need in this case.
+
+If a project is hosted on GitHub, it automatically provides basic stats (clones / repo visits) provided for all repos.
+
+#### Aggregated tracking
+
+This helps projects validate if the concept of what they're working on is valuable - do people install it? Do they actually use it at all?
+
+It also means they can get a sense of how accessible their software is by looking at how many people work out how to use it. At PostHog, early on, we tracked if people were able to get through the setup process. Many didn't, so we've kept tweaking it - and we still have work to do here :)
+
+With aggregated tracking, a project cannot by definition identify unique users. This means repeat visitors are likely to skew the statistics, and therefore, the understanding of user behavior.
+
+#### Anonymized user tracking
+
+This gives a much stronger signal than the above. It means you'll probably need to start using cookies or some way of obfuscating the user id from their behavior.
+
+It allows projects to see if people who started using the product are coming back, and which parts of the product they're returning too.
+
+#### Full user tracking
+
+Asking permission and not doing this by default feels like the only acceptable way from a user perspective.
+
+Tracking the actual users can really help understand the kinds of user profiles (ie if they work for big companies, or what kind of job roles they have) that find the software valuable. If a project is part of an open source company, then this is very likely to be of critical importance to them.
```

Comment:
I'd prefer these parentheses at the end of the sentence - it's disrupting the reading flow here